IntegrationTests/UI.IntegrationTests/CommandsDialogs/FormBrowseTests.cs

```diff
@@ -197,6 +191,77 @@ public void Filters_should_behave_as_expected()
                 });
         }
 
+        [Test]
+        public void ShowStashReference_should_filter_as_expected()
+        {
+            _referenceRepository.CreateCommit("Commit1");
+            _referenceRepository.Stash("Stash1");
+            _referenceRepository.CreateCommit("Commit2");
+            _referenceRepository.Stash("Stash2");
+            _referenceRepository.CreateCommit("Commit3");
+
+            bool showStashReference = AppSettings.ShowStashReference;
+            bool revisionGraphShowArtificialCommits = AppSettings.RevisionGraphShowArtificialCommits;
+
+            AppSettings.ShowStashReference = false;
+            AppSettings.RevisionGraphShowArtificialCommits = false;
+            RunFormTest(
+                form =>
+                {
+                    try
+                    {
+                        // 1. Check with ShowStashReference disabled
+                        Console.WriteLine("Scenario 1: set 'Show stash reference' to false");
+                        //// Assert
+                        AppSettings.ShowStashReference.Should().BeFalse();
+                        form.GetTestAccessor().RevisionGrid.GetTestAccessor().VisibleRevisionCount.Should().Be(4);
+
+                        // 2. Change ShowStashReference to enabled
+                        Console.WriteLine("Scenario 1: change 'Show stash reference' to enabled");
+                        form.GetTestAccessor().RevisionGrid.ToggleShowStashReference();
+                        //// Wait for reload to complete
+                        ProcessUntil(() => form.GetTestAccessor().RevisionGrid.GetTestAccessor().IsRefreshingRevisions.ToString(), false.ToString());
+                        //// Assert
+                        AppSettings.ShowStashReference.Should().BeTrue();
+                        form.GetTestAccessor().RevisionGrid.GetTestAccessor().VisibleRevisionCount.Should().Be(6);
+                    }
+                    finally
+                    {
+                        AppSettings.ShowStashReference = showStashReference;
+                        AppSettings.RevisionGraphShowArtificialCommits = revisionGraphShowArtificialCommits;
+                    }
+                });
+
+            AppSettings.ShowStashReference = true;
```

Comment:
Please break this into own test

Reply:
The tests needs a completelly clean repository, without other branches or stashes. ```reset --hard``` only cleanups the current branch.